card/lib/card/view_name.rb

```diff
@@ -0,0 +1,44 @@
+# -*- encoding : utf-8 -*-
+require_dependency 'card/env'
+
+require 'smart_name'
+
+class Card
+  class ViewName < SmartName
+    @@name2viewnameobject = {}
+
+    class << self
+      def new obj
+        return obj if self.class===obj
+        str = Array===obj ? obj*joint : obj.to_s
+        if known_name = @@name2viewnameobject[str]
+          known_name
+        else
+          super str.strip
+        end
+      end
+    end
+
+    def initialize str
+      @s = str.to_s.strip
+      @s = @s.encode('UTF-8') if RUBYENCODING
+      @key = if @s.index(self.class.joint)
+          @parts = @s.split(/\s*#{JOINT_RE}\s*/)
+          @parts << '' if @s[-1,1] == self.class.joint
+          @simple = false
+          @parts.map { |p| p.to_name.key } * self.class.joint
+        else
+          @parts = [str]
+          @simple = true
+          str.empty? ? '' : simple_key
+        end
+      @@name2viewnameobject[str] = self
+    end
+
+    def simple_key
+      decoded.underscore.gsub(/[^#{OK4KEY_RE}]+/,'_').split(/_+/).reject(&:empty?)*'_'
+    end
```

Comment:
Good thoughts.  Philipp, correct me if I'm wrong, but I think he mostly just wanted to change the name handling to get rid of some odd-looking names in code, as you say.  Specifically, he was having to do things like `default_edit_rule_args` when the view was `edit_rules`.

I couldn't immediately come up with (or remember) a strong argument for why view names *had* to be treated like cardnames, so it kind of came down to deciding which was the bigger wart: ugly method names or the ViewName code.

I do think the SmartName code could be made more adaptable, and that we'll probably do a better job of maintaining that once it's in the same repo.

In general it feels like we're losing a little bit of our simplicity in all these upgrades and may need to fight to gain it back.  The good news is that this isn't really affected much by data (unless there are references to different name variants in nest syntax in folks' cards), so we're a bit more pliant here.